src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,122 @@
 public class StringManipulation implements StringManipulationInterface {
 
+    private String string;
+
     @Override
     public String getString() {
-        return null;
+        return string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (string == null || string.isEmpty()) {
+            return 0;
+        }
+
+        String trimmedString = string.trim();
+        if (trimmedString.isEmpty()) {
+            return 0;
+        }
+
+        // Split the string by whitespace to get individual words
+        String[] words = trimmedString.split("\\s+");
+
+        return words.length;
     }
 
+
+
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
-    }
+    	 if (n <= 0) {
+             throw new IllegalArgumentException("Error: n is less than or equal to 0");
+         } 
+    	 else if (string == null) {
+             throw new NullPointerException("Error: String has not been initialized");
+         } 
+    	 else if (n > string.length()) {
+             throw new IndexOutOfBoundsException("Error: n exceeds string length");
+         }
+    	 
+    	 
+    	StringBuilder result = new StringBuilder();
+        int count = 0;  
+        for (int i = 0; i < string.length(); i++) {
+            char currentChar = string.charAt(i);
+            count++;
+                if (count % n == 0 && maintainSpacing) {
+                result.append(' ');
+            }    
+            if (count % n != 0) {
+                result.append(currentChar);
+            }
+
+
+        }
+        
+        return result.toString();
+        }
+
+
+
+
 
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        if (string == null) {
+        	throw new NullPointerException("Error: string is not yet initialized");
```

Comment:
Null Pointer Exception error isn't required for this method. I'm pretty sure this is actually supposed to be  an Illegal Argument Exception so I would modify this as part of that exception.